crates/relayer/src/chain/counterparty.rs

```diff
@@ -586,14 +592,17 @@ pub fn unreceived_acknowledgements(
     chain: &impl ChainHandle,
     counterparty_chain: &impl ChainHandle,
     path: &PathIdentifiers,
+    pagination: Paginate,
 ) -> Result<Option<(Vec<Sequence>, Height)>, Error> {
-    let (commitments_on_src, _) = commitments_on_chain(chain, &path.port_id, &path.channel_id)?;
+    let (commitments_on_src, _) =
+        commitments_on_chain(chain, &path.port_id, &path.channel_id, pagination.clone())?;
```

Comment:
`Paginate` can be made `Copy`:

```suggestion
        commitments_on_chain(chain, &path.port_id, &path.channel_id, pagination)?;
```